mlflow/gateway/handlers.py

```diff
@@ -0,0 +1,108 @@
+from enum import Enum
+from pathlib import Path
+from pydantic import BaseModel, Field, validator, parse_obj_as
+from pydantic.json import pydantic_encoder
+from typing import Optional, Union, List
+import yaml
+import json
+
+
+class Provider(str, Enum):
+    UNSPECIFIED_PROVIDER = "unspecified"
+    OPENAI = "openai"
+    ANTHROPIC = "anthropic"
+    BARD = "bard"
+
+
+class RouteType(str, Enum):
+    UNSPECIFIED = "unspecified"
+    LLM_V1_INSTRUCT = "llm/v1/instruct"
+    LLM_V1_CHAT = "llm/v1/chat"
+
+
+class OpenAIConfig(BaseModel):
+    openai_api_key: str = Field(..., alias="openai_api_key")
+    openai_api_type: Optional[str] = Field(None, alias="openai_api_type")
+    openai_api_base: Optional[str] = Field("https://api.openai.com/", alias="openai_api_base")
+    openai_api_version: Optional[str] = Field(None, alias="openai_api_version")
+
+
+class AnthropicConfig(BaseModel):
+    anthropic_api_key: str = Field(..., alias="anthropic_api_key")
+    anthropic_api_base: Optional[str] = Field(
+        "https://api.anthropic.com/", alias="anthropic_api_base"
+    )
+
+
+class BardConfig(BaseModel):
+    bard_api_key: str = Field(..., alias="bard_api_key")
+    bard_api_base: str = Field("https://bard.google.com/", alias="bard_api_base")
+
+
+class ModelInfo(BaseModel):
+    name: Optional[str] = None
+    provider: Provider = Provider.UNSPECIFIED_PROVIDER
+
+
+class Model(BaseModel):
+    name: Optional[str] = None
+    provider: Provider = Provider.UNSPECIFIED_PROVIDER
+    config: Optional[Union[OpenAIConfig, AnthropicConfig, BardConfig]] = None
+
+    @validator("provider", pre=True)
+    def validate_provider(cls, value):
+        if value in Provider._value2member_map_:
+            return value
+        return Provider.UNSPECIFIED_PROVIDER.value
+
+
+class RouteConfig(BaseModel):
+    name: Optional[str] = None
+    type: RouteType = RouteType.UNSPECIFIED
+    model: Optional[Model] = None
```

Comment:
Should we validate `model`? For example, if `model = gpt-6`, the service should fail to start.

Reply:
I'm all for it, but this might be a stretch goal for MVP. We could create a validation check where we take the API key, call a function that imports the python sdk and fetches the list of supported models in order to validate.